### PR TITLE
release/1.1.0: several bug fixes from installation of release/1.1.0 on HPCs, cloud, macOS

### PR DESCRIPTION
**Pull request by climbfuji**
### jedi-cmake

Turns out that `jedi-cmake@1.4.0` with recursive checkout doesn't install as it's currently written in the containers, because the containers somehow think they want to pull a `.tar.gz` file. The solution is to remove the `url` entry and specify only git commits, no `sha256` checksums.

This works on the containers and elsewhere and needs to be sent back to develop later.

###  crtm
Add comments in `crtm` package to switch to JCSDA-internal. These comments make it easier and faster to switch. This should stay in the release branch and not be merged back to develop.

Bug fix: `crtm` depends on `cmake@3.15` or later, this dependency was not captured in the package beforehand. This needs to be sent back to develop later.

### ncview
On some systems (e.g. Discover with Intel), `ncview` didn't find `udunits` even though the dependency was loaded by spack. This PR addresses that by explicitly passing the `udunits` `include` and `lib` dirs to `ncview`. This needs to be sent back to develop later.

### Enable CI tests for `release/**`

Enable CI tests for `release/**` - i.e. for this branch. Turns out that they don't run right now, however, because of Github action / spack (authoritative) changes beyond our control. This should be sent back to develop later for future use.

### CODE REVIEW

File: .github/workflows/unit_tests.yaml [hunk 1]
```diff
@@ -6,11 +6,13 @@ on:
       - develop
       - releases/**
       - jcsda_emc_spack_stack
+      - release/**
```
Comment (BenjaminTJohnson) [inline .github/workflows/unit_tests.yaml:9]: What's the difference between `release/` and `releases/` (other than `s`)?

**Review by BenjaminTJohnson — APPROVED** [review]

File: var/spack/repos/jcsda-emc/packages/jedi-cmake/package.py [hunk 2]
```diff
@@ -23,9 +23,6 @@ class JediCmake(CMakePackage):
         preferred=True,
         submodules=True,
     )
-    version("1.3.0", sha256="3e92339df858e9663b2cdd9f7bb7e56d18098e9c60606fe7af9e5f5911e5ca55")
-    version("1.2.0", sha256="eb9f1c403d1b43a90a5e774097382b183d56d5b40a1204b51af2da8db1559b21")
-    version("1.1.0", sha256="f1fe41eb5edd343bdf57eb76bea6d1b9f015878f0a9d0eb1e9dba18b903d3b35")
-    version("1.0.0", sha256="d773a800350e69372355b45e89160b593818cd438a86925b8a689c47996a0b9a")
+    version("1.3.0", commit="729a9b2ec97a7e93cbc58213493f28ca11f08754")
```
Comment (srherbener) [inline var/spack/repos/jcsda-emc/packages/jedi-cmake/package.py:26]: I'm not asking to change this, but for my own edification is this change necessary because of the switch to using a submodule in jedi-cmake? Thanks!

File: var/spack/repos/builtin/packages/crtm/package.py [hunk 3]
```diff
@@ -40,7 +43,7 @@ class Crtm(CMakePackage):
     # depends_on("ecbuild", when="@2.4.0:", type=("build"))
 
     # REL-2.4.0_emc (v2.4.0 ecbuild does not work)
-    version("2.4.0", commit="5ddd0d6")
+    version("2.4.0", commit="5ddd0d6")  # use commit="690da37" for JCSDA-internal
```
Comment (srherbener) [inline var/spack/repos/builtin/packages/crtm/package.py]: Would it be helpful to add a commented version statement with the JCSDA-internal commit hash? (Like the url's above).

File: .github/workflows/unit_tests.yaml (hunk 1 above)
Comment (climbfuji) [reply .github/workflows/unit_tests.yaml:9]: `releases/` is what spack (authoritative) is using, we use `release/`

File: var/spack/repos/jcsda-emc/packages/jedi-cmake/package.py (hunk 2 above)
Comment (climbfuji) [reply var/spack/repos/jcsda-emc/packages/jedi-cmake/package.py:26]: Yes, it is. Plus we don't ever plan to build older versions of jedi-cmake.

File: var/spack/repos/builtin/packages/crtm/package.py (hunk 3 above)
Comment (climbfuji) [reply var/spack/repos/builtin/packages/crtm/package.py]: Ok, can do.

Comment (climbfuji) [reply var/spack/repos/builtin/packages/crtm/package.py]: Done, please have a look

File: var/spack/repos/builtin/packages/ncview/package.py [hunk 4]
```diff
@@ -25,7 +25,13 @@ def configure_args(self):
 
         config_args = []
 
+        # Problems on some systems (e.g. NASA Discover with Intel)
+        # to find udunits include and library files despite
+        # dependency being specified above
+        config_args.append("--with-udunits2_incdir={}".format(spec["udunits"].prefix.include))
+        config_args.append("--with-udunits2_libdir={}".format(spec["udunits"].prefix.lib))
```
Comment (AlexanderRichert-NOAA) [inline var/spack/repos/builtin/packages/ncview/package.py:32]: Is the current issue with udunits not being found a deficiency in the udunits package/module, or an issue with how ncview is finding it?

Comment (climbfuji) [reply var/spack/repos/builtin/packages/ncview/package.py:32]: I don't know, `ncview` should find `udunits` just fine because the `depends_on(udunits)` is in the spack package, but for some reason on discover with Intel (and only there) it doesn't find it. Adding those two lines doesn't hurt. I verified that on the other systems, even without those two lines, `ncview` is finding the correct (spack) `udunits`, not some other installations in a default path.

Comment (AlexanderRichert-NOAA) [reply var/spack/repos/builtin/packages/ncview/package.py:32]: Do you know whether other packages that depend on it have had the same issue?

Comment (climbfuji) [reply var/spack/repos/builtin/packages/ncview/package.py:32]: No, I don't, and we don't have any other package in the stack that use udunits.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Review by srherbener — APPROVED** [review]
Thanks for addressing my comments!

File: var/spack/repos/jcsda-emc/packages/jedi-cmake/package.py (hunk 2 above)
Comment (srherbener) [reply var/spack/repos/jcsda-emc/packages/jedi-cmake/package.py:26]: Sounds good. Thanks!
